components/atom/tooltip/src/hoc/withIntersectionObserver.js

```diff
@@ -0,0 +1,39 @@
+/* global IntersectionObserver */
```

Comment:
This component is soooo interesting. :) We could adapt it as a standalone component in order to be reused in other components. For now, I would say is fine to be here, but let's keep it in mind.